services/web/server/src/simcore_service_webserver/login/_registration.py

```diff
@@ -0,0 +1,234 @@
+        raise web.HTTPConflict(
+            reason=cfg.MSG_PASSWORD_MISMATCH, content_type=MIMETYPE_APPLICATION_JSON
+        )
+
+    try:
+        parse_obj_as(EmailStr, email)
+    except ValidationError as err:
+        raise web.HTTPUnprocessableEntity(
+            reason="Invalid email", content_type=MIMETYPE_APPLICATION_JSON
+        ) from err
+
+    # NOTE: Extra requirements on passwords
+    # SEE https://github.com/ITISFoundation/osparc-simcore/issues/2480
+
+    if user := await db.get_user({"email": email}):
+        # Resets pending confirmation if re-registers?
+        if user["status"] == UserStatus.CONFIRMATION_PENDING.value:
+            _confirmation: ConfirmationTokenDict = await db.get_confirmation(
+                {"user": user, "action": ConfirmationAction.REGISTRATION.value}
+            )
+
+            if is_confirmation_expired(cfg, _confirmation):
+                await db.delete_confirmation(_confirmation)
+                await db.delete_user(user)
+                return
+
+        # If the email is already taken, return a 409 - HTTPConflict
+        raise web.HTTPConflict(
+            reason=cfg.MSG_EMAIL_EXISTS, content_type=MIMETYPE_APPLICATION_JSON
+        )
+
+    log.debug("Registration data validated")
+
+
+async def create_invitation_token(
+    db: AsyncpgStorage,
+    *,
+    user_id: IdInt,
+    user_email: Optional[EmailStr] = None,
+    tag: Optional[str] = None,
+    trial_days: Optional[PositiveInt] = None,
+) -> ConfirmationTokenDict:
+    """Creates an invitation token for a guest to register in the platform and returns
+
+        Creates and injects an invitation token in the confirmation table associated
+        to the host user
+
+    :param host: valid user that creates the invitation
+    :type host: Dict-like
+    :param guest: some description of the guest, e.g. email, name or a json
+    """
+    data_model = InvitationData.parse_obj(
+        {
+            "issuer": user_email,
+            "guest": tag,
+            "trial_account_days": trial_days,
+        }
+    )
+    confirmation = await db.create_confirmation(
+        user_id=user_id,
+        action=ConfirmationAction.INVITATION.name,
+        data=data_model.json(),
+    )
+    return confirmation
+
+
+async def check_invitation(
+    invitation_code: str, db: AsyncpgStorage, cfg: LoginOptions
+) -> InvitationData:
+    """
+    :raise web.HTTPForbidden if invalid
+    """
+    if confirmation := await validate_confirmation_code(invitation_code, db, cfg):
+        try:
+            invitation = _InvitationValidator.parse_obj(confirmation)
+            return invitation.data
+
+        except ValidationError as err:
+            log.warning(
+                "%s is associated with an invalid %s.\nDetails: %s",
+                f"{invitation_code=}",
+                f"{confirmation=}",
+                f"{err=}",
+            )
+
+        finally:
+            await db.delete_confirmation(confirmation)
```

Comment:
Having a delete in a method called check is a bit counter intuitive. I suppose you want to remove the confirmation if the confirmation is ok? Maybe a different function name might be more helpful. Maybe replace `check` with `validate` or something similar.

Reply:
i agree. will change by something like ``validate_and_remove_invitation`` thx